Core/GDCore/IDE/WholeProjectRefactorer.cpp

```diff
@@ -197,6 +206,57 @@ WholeProjectRefactorer::ComputeChangesetForVariablesContainer(
   return changeset;
 }
 
+std::shared_ptr<VariablesRenamingChangesetNode>
+WholeProjectRefactorer::ComputeChangesetForVariable(
+    const gd::Variable &oldVariable, const gd::Variable &newVariable) {
+
+  if (newVariable.GetChildrenCount() == 0 ||
+      oldVariable.GetChildrenCount() == 0) {
+    return std::shared_ptr<VariablesRenamingChangesetNode>(nullptr);
+  }
+
+  std::unordered_map<gd::String, gd::String> removedUuidAndNames;
+  for (const auto &pair : oldVariable.GetAllChildren()) {
+    const auto &oldName = pair.first;
+    const auto oldChild = pair.second;
+
+    // All variables are candidate to be removed.
+    removedUuidAndNames[oldChild->GetPersistentUuid()] = oldName;
+  }
+
+  auto changeset = std::make_shared<VariablesRenamingChangesetNode>();
+  for (const auto &pair : newVariable.GetAllChildren()) {
+    const auto &newName = pair.first;
+    const auto newChild = pair.second;
+
+    auto existingOldVariableUuidAndName =
+        removedUuidAndNames.find(newChild->GetPersistentUuid());
+    if (existingOldVariableUuidAndName == removedUuidAndNames.end()) {
+      // This is a new variable.
+      continue;
+    }
+    const gd::String &oldName = existingOldVariableUuidAndName->second;
+    const auto &oldChild = oldVariable.GetChild(oldName);
+
+    if (oldName != newName) {
+      // This is a renamed child.
+      changeset->oldToNewVariableNames[oldName] = newName;
+    }
+
+    const auto &childChangeset =
+        gd::WholeProjectRefactorer::ComputeChangesetForVariable(oldChild,
+                                                                *newChild);
+    if (childChangeset) {
+      changeset->modifiedVariables[oldName] = std::move(childChangeset);
+    }
+  }
```

Comment:
You never remove anything from `removedUuidAndNames`?

Reply:
It doesn't handle child-variables removing. After this PR, I'll remove the dead code for variable removing.
I should rename this to `oldVariableNamesByUuid`.

Reply:
Ok, maybe do the renaming then